modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisXMLDocumentConverterFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.postgis;
+
+import org.geotools.factory.Hints;
+import org.geotools.util.Converter;
+import org.geotools.util.XMLDocumentConverterFactory;
+import org.postgresql.util.PGobject;
+
+public class PostgisXMLDocumentConverterFactory extends
+        XMLDocumentConverterFactory {
+
+    @Override
+    public Converter createConverter(Class source, Class target, Hints hints) {
+        if (source.equals(PGobject.class)) {
```

Comment:
Ah ok, I did not notice that you're forcing the first argument to string, and then things work because the toString() of such pgObject happens to be the xml document itself.

This looks like a hack, generally speaking one cannot depend on toString() behavior, although in this particular case PGObject toString API state "the value of this object, in the syntax expected by org.postgresql"... so it might work fine, limited to PGObject.

I suggest you make a specific converter for PGObject instead of all these assumptions around the code that is supposed to be generic. Also, this would make it clearer what converter manages which conversions, right now you end up with the PostgisXMLDocumentConverterFactory being able to do the same conversions as the base class, which is also registered and available, plus one.